osu.Server.Queues.ScoreStatisticsProcessor/Commands/Maintenance/MigratePlaylistScoresToSoloScoresCommand.cs

```diff
@@ -0,0 +1,190 @@
+        /// The playlist room ID to reprocess.
+        /// </summary>
+        [Required]
+        [Argument(0, Description = "Command separated list of playlist room IDs to reprocess.")]
+        public string PlaylistIds { get; set; } = string.Empty;
+
+        public async Task<int> OnExecuteAsync(CancellationToken cancellationToken)
+        {
+            foreach (string id in PlaylistIds.Split(','))
+            {
+                if (cancellationToken.IsCancellationRequested)
+                    break;
+
+                using (var db = Queue.GetDatabaseConnection())
+                {
+                    var playlistItems = await db.QueryAsync<MultiplayerPlaylistItem>("SELECT * FROM multiplayer_playlist_items WHERE room_id = @PlaylistId", new
+                    {
+                        PlaylistId = int.Parse(id),
+                    });
+
+                    foreach (var item in playlistItems)
+                    {
+                        if (cancellationToken.IsCancellationRequested)
+                            break;
+
+                        MultiplayerScore[] scores = (await db.QueryAsync<MultiplayerScore>($"SELECT * FROM multiplayer_scores WHERE playlist_item_id = {item.id}")).ToArray();
+
+                        foreach (var score in scores)
+                        {
+                            Console.WriteLine($"Reprocessing score {score.id} from playlist {item.id}");
+
+                            Ruleset ruleset = LegacyRulesetHelper.GetRulesetFromLegacyId(item.ruleset_id);
+                            HitResult maxRulesetJudgement = ruleset.GetHitResults().First().result;
+                            Dictionary<HitResult, int> statistics = JsonConvert.DeserializeObject<Dictionary<HitResult, int>>(score.statistics)!;
+                            List<HitResult> allHits = statistics
+                                                      .SelectMany(kvp => Enumerable.Repeat(kvp.Key, kvp.Value))
+                                                      .ToList();
+                            var maximumStatistics = new Dictionary<HitResult, int>();
+
+                            foreach (var groupedStats in allHits
+                                                         .Select(r => getMaxJudgementFor(r, maxRulesetJudgement))
+                                                         .GroupBy(r => r))
+                            {
+                                maximumStatistics[groupedStats.Key] = groupedStats.Count();
+                            }
+
+                            APIMod[] roomMods = JsonConvert.DeserializeObject<APIMod[]>(item.required_mods)!;
+                            APIMod[] scoreMods = JsonConvert.DeserializeObject<APIMod[]>(score.mods)!;
+
+                            foreach (var m in roomMods)
+                                Debug.Assert(scoreMods.Contains(m));
+
+                            Console.WriteLine($"Mods: {string.Join(',', scoreMods.Select(m => m.ToString()))}");
```

Comment:
Was this intended to be in the command output? Feels kinda random.

Reply:
This was left over from the command I copied from because I was unsure mods were being reserialised correctly. Can be removed.